mslib/msui/mpl_qtwidget.py

```diff
@@ -1411,62 +1496,23 @@ def plot_kml(self, kmloverlay):
         """Apply settings from dictionary 'settings_dict' to the view.
 
         If settings is None, apply default settings.
         """
-        # logging.debug("applying map appearance settings %s." % settings)
-        settings = {"draw_graticule": True,
-                    "draw_coastlines": True,
-                    "fill_waterbodies": True,
-                    "fill_continents": True,
-                    "draw_flighttrack": True,
-                    "draw_marker": True,
-                    "label_flighttrack": True,
-                    "tov_plot_title_size": "default",
-                    "tov_axes_label_size": "default",
-                    "colour_water": ((153 / 255.), (255 / 255.), (255 / 255.), (255 / 255.)),
-                    "colour_land": ((204 / 255.), (153 / 255.), (102 / 255.), (255 / 255.)),
-                    "colour_ft_vertices": (0, 0, 1, 1),
-                    "colour_ft_waypoints": (1, 0, 0, 1)}
-        if settings_dict is not None:
-            settings.update(settings_dict)
-
-        # Stores the exact value of fontsize for topview plot title size(tov_pts)
-        self.tov_pts = (self.topview_size_settings["plot_title_size"] if settings["tov_plot_title_size"] == "default"
-                        else int(settings["tov_plot_title_size"]))
-        # Stores the exact value of fontsize for topview axes label size(tov_als)
-        self.tov_als = (self.topview_size_settings["axes_label_size"] if settings["tov_axes_label_size"] == "default"
-                        else int(settings["tov_axes_label_size"]))
-
-        self.appearance_settings = settings
-        ax = self.ax
-
-        if self.map is not None:
-            self.map.set_coastlines_visible(settings["draw_coastlines"])
-            self.map.set_fillcontinents_visible(visible=settings["fill_continents"],
-                                                land_color=settings["colour_land"],
-                                                lake_color=settings["colour_water"])
-            self.map.set_mapboundary_visible(visible=settings["fill_waterbodies"],
-                                             bg_color=settings["colour_water"])
-            self.waypoints_interactor.set_path_color(line_color=settings["colour_ft_vertices"],
-                                                     marker_facecolor=settings["colour_ft_waypoints"])
-            self.waypoints_interactor.show_marker = settings["draw_marker"]
-            self.waypoints_interactor.set_vertices_visible(settings["draw_flighttrack"])
-            self.waypoints_interactor.set_labels_visible(settings["label_flighttrack"])
-
-            # Updates plot title size as selected from combobox labelled plot title size.
-            ax.set_autoscale_on(False)
-            ax.set_title("Top view", fontsize=self.tov_pts, horizontalalignment="left", x=0)
+        self.myfig.set_map_appearance(settings_dict)
+        if self.waypoints_interactor is not None:
```

Comment:
I left this here to avoid touching the waypoints_interactor. This might require further changes, once we start adressing refactoring the flightpath.